lib/lcdbwf/tests/testthat/test-contrasts.R

```diff
@@ -0,0 +1,246 @@
+  }) # test_that
+} # for type
+# ---------------------------------------------------------------- #
+
+
+# -------------- missing both test and type with LRT ------------- #
+# Similar structure to the ASCII table depicted tests from above
+# with the 'test' argument also missing in make_results
+# and test also missing. When test is missing, test should be detected from dds_lrt as 'LRT'.
+# With type also missing, type should be set as the current default: 'ashr' as of 5-30-2024.
+# This combination of test and type is incompatible and so the following error message should
+# be returned.
+test_that("make_results errors when user attempts to run lfcShrink by defining a missing type when test == 'LRT'", {
+  expect_error(make_results(dds_name='dds_lrt',
+                            label='missing test and type of LRT DDS'),
+                            "You cannot pass a non-NULL or missing type to make_results with an LRT dds object. For LRT, LFC values are set to 0 and should not be passed to lfcShrink. Use type == NULL in make_results for LRT DDS objects.")
+}) # test_that
+# ---------------------------------------------------------------- #
+
+
+# ------------- test included for all types dds Wald ------------- #
+# Similar structure to the ASCII table depicted tests from above
+# but with the 'test' argument included in make_results
+# and test == 'Wald'
+for (type in c('ashr', 'apeglm', 'normal')) {
+  test_that("make_results returns a DESeqResults object with !all res$res$LFC == 0 when user passes a defined type along with test == 'Wald'", {
+    if (type != 'apeglm') {
+      res <- make_results(dds_name='dds_wald',
+                          label='Shrink Wald results',
+                          contrast=contrast,
+                          type=type,
+                          test='Wald')
+      expect_true(is_deseq_res(res$res))
+      expect_true(identical(names(res), c('res', 'dds', 'label')))
+      expect_true(!all(res$res$log2FoldChange == 0))
+      expect_true(any(grepl('Wald', mcols(res$res)$description[4])))
+    } else if (type == 'apeglm') {
+      res <- make_results(dds_name='dds_wald',
+                                        label='Shrink Wald results',
+                                        coef=coef,
+                                        type=type,
+                                        test='Wald')
+      expect_true(is_deseq_res(res$res))
+      expect_true(identical(names(res), c('res', 'dds', 'label')))
+      expect_true(!all(res$res$log2FoldChange == 0))
+      expect_true(any(grepl('Wald', mcols(res$res)$description[4])))
+    } # if type
+  }) # test_that
+} # for type
+# ---------------------------------------------------------------- #
+
+
+# ---------------------- missing dds_list ------------------------ #
+remove(dds_list)
```

Comment:
might want to add it back in later after this test, otherwise adding new tests to the bottom of this file in the future may fail.

Reply:
addressed in 9828f4690e7bdee5